From e501953c440e2dd0cd11b9707bf5f9d157ba74a4 Mon Sep 17 00:00:00 2001
From: Neha Malcom Francis <n-francis@ti.com>
Date: Fri, 4 Jul 2025 12:50:36 +0530
Subject: [PATCH] kconfig: Add support for external config fragment and
 defconfig paths

Add support for fetching config fragments and defconfigs externally using
CONFIG_FRAGMENT_PATH and CONFIG_DEFCONFIG_PATH respectively.

Upstream-Status: Pending

Signed-off-by: Neha Malcom Francis <n-francis@ti.com>
---
 scripts/kconfig/Makefile | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/scripts/kconfig/Makefile b/scripts/kconfig/Makefile
index f0f93c56bdb..2db3940542a 100644
--- a/scripts/kconfig/Makefile
+++ b/scripts/kconfig/Makefile
@@ -12,6 +12,14 @@ PHONY += xconfig gconfig menuconfig config localmodconfig localyesconfig \
 #  Set SRCARCH to .. fake this Makefile.
 SRCARCH := ..
 
+# If a config fragment file is kept in another path, add it to the list of paths
+# to be searched within
+CONFIG_FRAGMENT_PATH ?= $(srctree)/arch/$(SRCARCH)/configs
+
+# If a defconfig file is kept in another path, add it to the list of paths
+# to be searched
+CONFIG_DEFCONFIG_PATH ?= $(srctree)/arch/$(SRCARCH)/configs
+
 ifdef KBUILD_KCONFIG
 Kconfig := $(KBUILD_KCONFIG)
 else
@@ -93,7 +101,7 @@ endif
 endif
 
 %_defconfig: $(obj)/conf
-	$(Q)$(CPP) -nostdinc -P -I $(srctree) -undef -x assembler-with-cpp $(srctree)/arch/$(SRCARCH)/configs/$@ -o generated_defconfig
+	$(Q)$(CPP) -nostdinc -P -I $(srctree) -undef -x assembler-with-cpp $(CONFIG_DEFCONFIG_PATH)/$@ -o generated_defconfig
 	$(Q)$< $(silent) --defconfig=generated_defconfig $(Kconfig)
 
 # Added for U-Boot (backward compatibility)
@@ -103,6 +111,7 @@ endif
 
 configfiles=$(wildcard $(srctree)/kernel/configs/$@ \
 		$(srctree)/arch/$(SRCARCH)/configs/$@ \
+		$(CONFIG_FRAGMENT_PATH)/$@ \
 		$(shell find $(srctree)/board -name "$@"))
 
 %.config: $(obj)/conf
-- 
2.17.1

